tools/base/libboard/board_vc/board_vc.c

```diff
@@ -0,0 +1,749 @@
+	}
+
+
+	/* BMC FW version format reading
+	NIOS II Firmware Build 0x0 32 RW[23:0] 24 hFFFFFF Build version of NIOS II Firmware
+	NIOS FW is up e.g. 1.0.1 for first release
+	[31:24] 8hFF Firmware Support Revision - ASCII code
+	0xFF is the default value without NIOS FW, will be changed after NIOS FW is up
+	0x41(A)-For RevA
+	0x42(B)-For RevB
+	0x43(C)-For RevC
+	0x44(D)-For RevD
+	*/
+
+	var = strtoul(buf, NULL, VER_BUF_SIZE);
+
+	rev = (var >> 24) & 0xff;
+	if ((rev > 0x40) && (rev < 0x5B)) {// range from 'A' to 'Z'
+		snprintf_s_ciii(fw_ver, len, "%c.%u.%u.%u", (char)rev, (var >> VER_BUF_SIZE) & 0xff, (var >> 8) & 0xff, var & 0xff);
+	}
+	else {
+		OPAE_ERR("Invalid BMC firmware version");
+		res = FPGA_EXCEPTION;
+	}
+
+	return res;
+}
+
+// Read MAX10 firmware version
+fpga_result read_max10fw_version(fpga_token token, char *max10fw_ver, size_t len)
+{
+	fpga_result res                      = FPGA_OK;
+	fpga_result resval                   = FPGA_OK;
+	uint32_t size                        = 0;
+	char buf[VER_BUF_SIZE]               = { 0 };
+	fpga_object max10fw_object;
+
+	if (max10fw_ver == NULL) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_MAX10_VER, &max10fw_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token object");
+		return res;
+	}
+
+	res = fpgaObjectGetSize(max10fw_object, &size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	res = fpgaObjectRead(max10fw_object, (uint8_t *)buf, 0, size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	// Return error if input sysfs path length smaller then object size
+	if (len < size) {
+		FPGA_ERR("Invalid Input sysfs path size");
+		resval = FPGA_EXCEPTION;
+		goto out_destroy;
+	}
+
+	res = parse_fw_ver(buf, max10fw_ver, len);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to parse version ");
+		resval = res;
+		goto out_destroy;
+	}
+
+out_destroy:
+	res = fpgaDestroyObject(&max10fw_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+	return resval;
+}
+
+// Read PCB information
+fpga_result read_pcb_info(fpga_token token, char *pcb_info, size_t len)
+{
+	fpga_result res                = FPGA_OK;
+	fpga_result resval             = FPGA_OK;
+	uint32_t size                  = 0;
+	fpga_object pcb_object;
+
+	if (pcb_info == NULL ) {
+		FPGA_ERR("Invalid input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_PCB_INFO, &pcb_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token Object");
+		return res;
+	}
+
+	res = fpgaObjectGetSize(pcb_object, &size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object size");
+		resval = res;
+		goto out_destroy;
+	}
+
+	// Return error if input sysfs path length smaller then object size
+	if (len < size) {
```

Comment:
I don't think this check is sufficient (or the comment is misleading). The comment shouldn't refer to a "sysfs path length".

Reply:
Fixed